src/prefect/flows.py

```diff
@@ -158,6 +159,34 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hook passed is list and contains callables
+        hook_categories = [on_completion, on_failure, on_cancellation, on_crashed]
+        hook_names = ["on_completion", "on_failure", "on_cancellation", "on_crashed"]
+        for hooks, hook_name in zip(hook_categories, hook_names):
+            if hooks is not None:
+                if not hooks:
+                    raise ValueError(f"Empty list passed for '{hook_name}'")
+                try:
+                    hooks = list(hooks)
+                except TypeError:
+                    raise TypeError(
+                        f"Expected iterable for '{hook_name}'; got"
+                        f" {type(hooks).__name__} instead. Please provide a list of"
+                        f" hooks to '{hook_name}'. For"
+                        f" example:\n@flow({hook_name}=[hook1, hook2])\ndef"
+                        " flow1():\npass"
+                    )
```

Comment:
```suggestion
                    raise TypeError(
                        f"Expected iterable for '{hook_name}'; got"
                        f" {type(hooks).__name__} instead. Please provide a list of"
                        f" hooks to '{hook_name}':\n\n"
                        f"@flow({hook_name}=[hook1, hook2])\ndef"
                        " my_flow():\n\tpass"
                    )
```
Let's add indentation here, it will look nicer when it prints out:

```bash
  File "/Users/bean/code/prefect/src/prefect/flows.py", line 172, in __init__
    raise TypeError(
TypeError: Expected iterable for 'on_completion'; got function instead. Please provide a list of hooks to 'on_completion':

@flow(on_completion=[hook1, hook2])
def my_flow():
        pass
```

Reply:
Yeah it looks great, on it.